central/graphql/resolvers/image_vulnerabilities_test.go

```diff
@@ -0,0 +1,364 @@
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCountScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	count, err := image.ImageVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(3), count)
+
+	image = s.getImageResolver(ctx, "sha2")
+
+	count, err = image.ImageVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(5), count)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCounter() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	count, err := s.resolver.ImageVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), count, 5, 1, 1, 2, 1, 1)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityCounterScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	count, err := image.ImageVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), count, 3, 1, 1, 0, 1, 1)
+
+	image = s.getImageResolver(ctx, "sha2")
+
+	count, err = image.ImageVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), count, 5, 1, 1, 2, 1, 1)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestTopImageVulnerabilityUnscoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	_, err := s.resolver.TopImageVulnerability(ctx, RawQuery{})
+	s.Error(err)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestTopImageVulnerability() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	image := s.getImageResolver(ctx, "sha1")
+
+	_, err := image.TopImageVulnerability(ctx, RawQuery{})
+	s.NoError(err)
+
+	// TODO figure out how to test this
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityImages() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vuln := s.getImageVulnerabilityResolver(ctx, "cve-2018-1#")
+
+	images, err := vuln.Images(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(2, len(images))
+
+	vuln = s.getImageVulnerabilityResolver(ctx, "cve-2017-1#")
+
+	images, err = vuln.Images(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(1, len(images))
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityImageCount() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vuln := s.getImageVulnerabilityResolver(ctx, "cve-2018-1#")
+
+	count, err := vuln.ImageCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(2), count)
+
+	vuln = s.getImageVulnerabilityResolver(ctx, "cve-2017-1#")
+
+	count, err = vuln.ImageCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(1), count)
+}
+
+func (s *GraphQLImageVulnerabilityTestSuite) TestImageVulnerabilityImageComponents() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vuln := s.getImageVulnerabilityResolver(ctx, "cve-2018-1#")
+
+	images, err := vuln.ImageComponents(ctx, PaginatedQuery{})
```

Comment:
typo

Reply:
I don't think this is a typo, we're testing that vuln's subresolver ImageComponents is returning data we expect. This is maybe something better done in the ImageComponent test file, but the test holds true all the same

Reply:
`images` -> `imageComponents`